sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -56,11 +58,21 @@ export class WorkloadIdentityCredential implements TokenCredential {
     const workloadIdentityCredentialOptions = options ?? {};
     const tenantId = workloadIdentityCredentialOptions.tenantId || process.env.AZURE_TENANT_ID;
     const clientId = workloadIdentityCredentialOptions.clientId || process.env.AZURE_CLIENT_ID;
+
     this.federatedTokenFilePath =
-      workloadIdentityCredentialOptions.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+      workloadIdentityCredentialOptions?.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+    this.serviceConnectionId = workloadIdentityCredentialOptions.serviceConnectionId;
+
     if (tenantId) {
       checkTenantId(logger, tenantId);
     }
+    if (this.federatedTokenFilePath && this.serviceConnectionId) {
+      const errorMessage = `${credentialName}: ambiguous. serviceConnectionId and federatedTokenFilePath cannot be provided at the same time. These are used for supporting WorkloadIdentity for different environments. 
```

Comment:
I think `federatedTokenFilePath` is what we name the variable internally but the param is called `tokenFilePath` - if that's the case we should change the error message to specify the correct param name (otherwise it may be confusing to the user).

Also, is it an error to have AZURE_FEDERATED_TOKEN_FILE env var set and pass in `serviceConnectionId` via the params? Just curious

Reply:
Yes that is going to be an error scenario.